purview/utils/general.py

```diff
@@ -183,6 +185,12 @@ def query_neo4j(query):
     if not results:
         return results_dict
 
+    if isinstance(results[0][0], int):
+        return results[0][0]-1
+
+    if isinstance(results[0][0], Path):
```

Comment:
Add a comment above saying that if a Path is in the first result, we assume that's all we want to process.